cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -228,8 +233,7 @@ private void maybeAddLeaderChangeTasks(Collection<ExecutionProposal> proposals,
             // involved in this round.
             int sourceBroker = task.proposal().oldLeader();
             Set<Integer> destinationBrokers = task.proposal().replicasToAdd();
-            if (brokerInvolved.contains(sourceBroker)
-                || KafkaCruiseControlUtils.containsAny(brokerInvolved, destinationBrokers)) {
```

Comment:
It feels like removing this line allows brokers to be both source and destination brokers? Not "allows different source brokers for the same destination brokers"?

Reply:
This improvement helps when replacing a failed broker. In this case, there's usually only one destination broker (the broker which we add in order to replace the failed one).
As you can see in the code when the movement task is being added to the batch, its source broker and destination brokers are being added to the set of "involved brokers". 
This check doesn't allow (skip) any other movement tasks that have source or destination brokers already contained in that "involved brokers" set.
So, generally, there is an outer `while` loop, it clears "involved brokers" set.
And then there is an inner `for` loop which iterates over all brokers and tries to get next available task from each broker and add it to the list.
But since all brokers have tasks with the same destination broker, only tasks from the first broker would get added to the list due to this check.

Without this relaxation, the rebalancing process on replacing a failed broker performs tasks only by one broker at a time.  Which also impacts on the speed. We want to fetch data ffrom different source brokers in order to increase overall fetch speed.

Reply:
> But since all brokers have tasks with the same destination broker, only tasks from the first broker would get added to the list due to this check.

It sounds like "allow multi source to multi destination" then "allow multiple source to the same destination"?